contracts/sol6/wrappers/KyberRateHelper.sol

```diff
@@ -71,6 +70,45 @@ contract KyberRateHelper is IKyberRateHelper, WithdrawableNoModifiers, Utils5 {
         return getRatesForTokenWithCustomFee(token, optionalBuyAmount, optionalSellAmount, 0);
     }
 
+    /// @dev function to cover backward compatible with old network interface
+    /// @dev get rate from token to eth, use the best eth amount to get rate from eth to token
+    /// @param token Token to get rate
+    /// @param optionalAmountWei Amount to get rate (default: 0)
+    function getReservesRates(IERC20 token, uint256 optionalAmountWei)
+        public
+        override
+        view
+        returns (
+            bytes32[] memory buyReserves,
+            uint256[] memory buyRates,
+            bytes32[] memory sellReserves,
+            uint256[] memory sellRates
+        )
+    {
+        (uint256 networkFeeBps, ) = kyberDao.getLatestNetworkFeeData();
+        uint256 buyAmount = optionalAmountWei > 0 ? optionalAmountWei : 1 ether;
+
+        (buyReserves, buyRates) = getBuyInfo(token, buyAmount, networkFeeBps);
+
+        uint256 bestRate = 0;
+        for (uint256 i = 0; i < buyRates.length; i++) {
+            if (buyRates[i] > bestRate) {
+                bestRate = buyRates[i];
+            }
+        }
+
+        if (bestRate == 0) {
```

Comment:
so if we don't have buy rate, then sell rate will also be zero?

Reply:
for now yes.
this way we have strict API
if later its a problem we can update.